File: C++/Lezione 21.cpp
//
//  main.cpp
//  Lezione 21
//
//  Created by Dario Ciuciulla on 27/10/25.
// Selection Sort
//Decrescente
/*
#include <iostream>
using namespace std;

int a[8]= {41, 37, 10, 74, 98, 22, 83, 66};
int n=8;
int main(){
    
    for(int i=(n-1); i>=0; i--){
        int max = i;
        for(int j=i-1; j>=0; j--){
            if(a[j]>a[max])
                max=j;
        }
        if(max!=i){
            int temp = a[i];
            a[i]=a[max];
            a[max]=temp;
        }
        cout<<a[i]<<" ";
    }
    
   
}
*/
//Crescente

#include <iostream>
using namespace std;

int a[8]= {41, 37, 10, 74, 98, 22, 83, 66};

int main(){
    
    for(int i=0; i<7; i++){
        int min = i;
        for(int j=i+1; j<7; j++){
            if(a[j]<a[min])
                min=j;
        }
        if(min!=i){
            int temp = a[i];
            a[i]=a[min];
            a[min]=temp;
        }
        cout<<a[i]<<" ";
    }
    
   
}

File: Python/Lezione 21.py
#Selection Sort
# Decrescente
"""
a=[41, 37, 10, 74, 98, 22, 83, 66]
n=8

for i in range(n-1,-1,-1):
    max=i
    for j in range(i-1,-1,-1):
        if a[j]<a[max]:
            max=j
    if max!=i:
        temp=a[i]
        a[i]=a[max]
        a[max]=temp

print(a)
"""
#Crescente
a=[41, 37, 10, 74, 98, 22, 83, 66]
n=8

for i in range(0,n,+1):
    min=i
    for j in range(i+1,n,+1):
        if a[j]<a[min]:
            min=j
    if min!=i:
        temp=a[i]
        a[i]=a[min]
        a[min]=temp

print(a)

